components/brave_vpn/brave_vpn_unittest.cc

```diff
@@ -230,168 +307,164 @@ TEST(BraveVPNFeatureTest, DISABLED_FeatureTest) {
-  // pref_service_.SetBoolean(skus::prefs::kSkusVPNHasCredential, true);
-  service_->OnFetchHostnames("region-a", GetHostnamesData(), true);
-  EXPECT_EQ(ConnectionState::CONNECTING, service_->connection_state_);
+  connection_state() = ConnectionState::CONNECTING;
+  OnFetchHostnames("region-a", GetHostnamesData(), true);
+  EXPECT_EQ(ConnectionState::CONNECTING, connection_state());
 
   // To prevent real os vpn entry creation.
-  service_->is_simulation_ = true;
-  service_->OnGetProfileCredentials(GetProfileCredentialData(), true);
-  EXPECT_EQ(ConnectionState::CONNECTING, service_->connection_state_);
-  EXPECT_TRUE(service_->connection_info_.IsValid());
+  is_simulation() = true;
+  OnGetProfileCredentials(GetProfileCredentialData(), true);
+  EXPECT_EQ(ConnectionState::CONNECTING, connection_state());
+  EXPECT_TRUE(GetConnectionInfo().IsValid());
 
   // Check cached connection info is cleared when user set new selected region.
-  service_->connection_state_ = ConnectionState::DISCONNECTED;
-  brave_vpn::mojom::Region region;
-  service_->SetSelectedRegion(region.Clone());
-  EXPECT_FALSE(service_->connection_info_.IsValid());
+  connection_state() = ConnectionState::DISCONNECTED;
+  service_->SetSelectedRegion(brave_vpn::mojom::Region().Clone());
+  EXPECT_FALSE(GetConnectionInfo().IsValid());
 }
 
 // TODO(bsclifton): re-enable test after figuring out why crash is happening
 TEST_F(BraveVPNServiceTest, DISABLED_NeedsConnectTest) {
   // Check ignore Connect() request while connecting or disconnecting is
   // in-progress.
-  service_->connection_state_ = ConnectionState::CONNECTING;
-  service_->Connect();
-  EXPECT_EQ(ConnectionState::CONNECTING, service_->connection_state_);
+  connection_state() = ConnectionState::CONNECTING;
+  Connect();
+  EXPECT_EQ(ConnectionState::CONNECTING, connection_state());
 
-  service_->connection_state_ = ConnectionState::DISCONNECTING;
-  service_->Connect();
-  EXPECT_EQ(ConnectionState::DISCONNECTING, service_->connection_state_);
+  connection_state() = ConnectionState::DISCONNECTING;
+  Connect();
+  EXPECT_EQ(ConnectionState::DISCONNECTING, connection_state());
 
   // Handle connect after disconnect current connection.
-  service_->connection_state_ = ConnectionState::CONNECTED;
-  service_->Connect();
-  EXPECT_TRUE(service_->needs_connect_);
-  EXPECT_EQ(ConnectionState::DISCONNECTING, service_->connection_state_);
-  service_->OnDisconnected();
-  EXPECT_FALSE(service_->needs_connect_);
-  EXPECT_EQ(ConnectionState::CONNECTING, service_->connection_state_);
+  connection_state() = ConnectionState::CONNECTED;
+  Connect();
+  EXPECT_TRUE(needs_connect());
+  EXPECT_EQ(ConnectionState::DISCONNECTING, connection_state());
+  OnDisconnected();
+  EXPECT_FALSE(needs_connect());
+  EXPECT_EQ(ConnectionState::CONNECTING, connection_state());
 }
 
 // TODO(bsclifton): re-enable test after figuring out why crash is happening
 TEST_F(BraveVPNServiceTest, DISABLED_LoadRegionDataFromPrefsTest) {
```

Comment:
why these tests are disabled? why we're not fixing them? a crash in test is a possible crash in production.

Reply:
Working on it. I'll enable all and increase more test coverage in this PR.